### PR TITLE
Set File Types to Sync from Obsidian via Khoj Plugin Settings Page

### LINKED ISSUE #893: [FIX] Allow Setting File Types to Sync from Obsidian
## Describe the bug
i get this eveytime i click update or restart "Failed to sync your content with Khoj server.Requests were throttled. Upgrade your subscription or try again "

## Screenshots
Xnapper-2024-08-20-23 41 30

## Platform
- Server:
    - [x] Cloud-Hosted (https://app.khoj.dev)
    - [ ] Self-Hosted Docker
    - [ ] Self-Hosted Python package
    - [ ] Self-Hosted source code
- Client:
    - [x] Obsidian
    - [ ] Emacs
    - [ ] Desktop app
    - [ ] Web browser
    - [ ] WhatsApp
- OS:
    - [ ] Windows
    - [x] macOS
    - [ ] Linux
    - [ ] Android
    - [ ] iOS

### If self-hosted
- Server Version [e.g. 1.0.1]:

### PR DESCRIPTION
**Pull request by debanjum**
Limit file types to sync with Khoj from Obsidian to:
- Avoid hitting per user index-able data limits, especially for folks on the Khoj cloud free tier. E.g by excluding images in Obsidian vault from being synced
- Improve context used by Khoj to generate responses

Khoj_Obsidian_Sync_Files_Types_Settings

Closes #893

### CODE REVIEW

File: src/interface/obsidian/src/settings.ts [hunk 1]
```diff
@@ -141,9 +189,8 @@ export class KhojSettingTab extends PluginSettingTab {
                     this.plugin.registerInterval(progress_indicator);
 
                     this.plugin.settings.lastSync = await updateContentIndex(
-                        this.app.vault, this.plugin.settings, this.plugin.settings.lastSync, true
+                        this.app.vault, this.plugin.settings, this.plugin.settings.lastSync, true, true
                     );
-                    new Notice('✅ Updated Khoj index.');
```
Comment (sabaimran) [inline src/interface/obsidian/src/settings.ts:146]: Should we leave this notice here? Can be a helpful indicator to the user.

File: src/interface/obsidian/src/utils.ts [hunk 2]
```diff
@@ -48,14 +48,29 @@ function filenameToMimeType (filename: TFile): string {
...
-
-export async function updateContentIndex(vault: Vault, setting: KhojSetting, lastSync: Map<TFile, number>, regenerate: boolean = false): Promise<Map<TFile, number>> {
+export const fileTypeToExtension = {
+    'pdf': ['pdf'],
+    'image': ['png', 'jpg', 'jpeg'],
+    'markdown': ['md', 'markdown'],
+};
+export const supportedImageFilesTypes = fileTypeToExtension.image;
+export const supportedBinaryFileTypes = fileTypeToExtension.pdf.concat(supportedImageFilesTypes);
+export const supportedFileTypes = fileTypeToExtension.markdown.concat(supportedBinaryFileTypes);
```
Comment (sabaimran) [inline src/interface/obsidian/src/utils.ts:58]: Where is the user setting being applied? It seems like it's just using all the allowed file types and not filtering anything out.

Comment (sabaimran) [reply src/interface/obsidian/src/utils.ts:58]: Ah nevermind, it's happening below with the use of the `KhojSetting` object.

File: src/interface/obsidian/src/utils.ts [hunk 3]
```diff
@@ -134,12 +179,20 @@ export async function updateContentIndex(vault: Vault, setting: KhojSetting, las
     if (error_message) {
         new Notice(error_message);
     } else {
+        if (userTriggered) new Notice('✅ Updated Khoj index.');
```
Comment (sabaimran) [inline src/interface/obsidian/src/utils.ts:182]: Ah, seems like this would better capture the different indexing pathways.

**Review by sabaimran — APPROVED** [review]
Awesome, appreciate the helpful screenshot also.

File: src/interface/obsidian/src/settings.ts (hunk 1 above)
Comment (debanjum) [reply src/interface/obsidian/src/settings.ts:146]: Yeah, it's been moved into the index updater function. We now only show the indexing success notice when manually triggered by user and indexing succeeded. Previously we'd show indexing success notice even on error, after the error notice which was weird

File: src/interface/obsidian/src/utils.ts (hunk 3 above)
Comment (debanjum) [reply src/interface/obsidian/src/utils.ts:182]: Yeah, putting it here allows only showing the indexing success notice if not error and user triggered indexing manually (e.g via Khoj obsidian plugin settings page)

**Pull request merged**